### ISSUE TITLE
1.14 mobile release

### ISSUE THREAD

**Issue by churik** [status-im/status-react#12184]
issue to manage mobile 1.14 mobile release

### Scope
https://github.com/status-im/status-react/milestone/53

- release notes: https://notes.status.im/1.14-release-notes?view
- release branch: https://github.com/status-im/status-react/tree/release/1.14.x

### Testing info 
Main test suite: https://ethstatus.testrail.net/index.php?/runs/view/7440
Upgrade (e2e): https://ethstatus.testrail.net/index.php?/runs/view/7441

### Builds in TestFlight and PlayStore to push as a release
https://ci.status.im/job/status-react/job/release/job/release%252F1.14.x/3/
APK:  https://status-im-releases.ams3.digitaloceanspaces.com/StatusIm-Mobile-v1.14.0-d1a3a2-universal.apk
IOS: 1.14.0(20210607123628)

#### 1) Kick-off
- [x] 1. Translation PRs is merged before cutting branch (Jinho)
- [x] 2. Feature or improvements release? Prepare scope of visible changes based on commits history, started from previous release branch cut (Feature release needs more lead time and assets for comms, [example]( https://notes.status.im/1.14-release-notes?view))
Mostly improvements, based on 
- [x] 3. Create 1.14.0 release branch and get successful builds (doc is [here](https://github.com/status-im/status-react/blob/develop/doc/RELEASE_GUIDE.md) so, testing can be started)
- [x] 4. Based on release scope, ask for comms (Jonny) 
- [x] 5. Make sure that assets in stores are up to date. If should be updated, ask @cammellos  for help

#### 2) Testing
- [x] Dogfooding (add notes, if it was held)
- [x] Regression testing for release candidate build
- [x] Upgrade tests
- [x] Fix known blocker(s) found during regression testing

##### Blockers
- [x] 1 blocker: https://github.com/status-im/status-react/issues/12191

=============================================================================================

- [x] Cherry-pick fix(es) to the release branch
- [x] Verify fix(es) on new release candidate build
- [x] Retest
- [x] Battery consumption testing (if makes sense)
- [x] Keep release build forever (button in Jenkins)

#### 3) Submission

- [x] 1. Release notes written and added (500 character limit for updates (Android), no Android references for App store(should be available from set 1.4)
- [x] 2. Submit beta version for Android (Open Testing in Play Store)
- [x] 3. Submit iOS build to Apple review ([instruction](https://drive.google.com/file/d/10Cl7PBB7TFPkZiVbfzdFGpfMRP9bxXuq/view?usp=sharing), be careful - low quality of audio)
- [x] 4. Play store content reviewed and updated
- [x] 5. Privacy policy reviewed and updated (ask in channel)
- [x] 6. Countries to be excluded (ask in channel) 
- [x] 7. Changes to our FAQ made (should be checked and if any changes needed, ask Jakub to made them)
- [x] 8. Prepare PR for F-droid build ([instruction]( https://github.com/status-im/status-react/blob/develop/doc/FDROID.md))

#### 4) Before publishing
- [x] 1. Translations for comms available (marketing, ask in the channel)
- [x] 2. Promotional content ready (marketing, ask in channel)
- [x]  3. Upload builds for DO store,  https://dev.status.im/ (Jakub)

#### 5) At time of publishing
- [x] 1. notify marketing to push comms and blog post (Jonny)
- [x] 2. Publish release to app store and playstore
- [x] 3. Final push to update prod site https://status.im/,  [instructions](https://github.com/status-im/infra-docs/blob/master/articles/status_release_upload.md) (Jakub - requires credentials to upload) 
- [x] 4. Add the link to the post about the release and publish the release: https://github.com/status-im/status-react/releases (Jakub)
- [x] 5. update https://github.com/status-im/status-react/blob/develop/RELEASES.md with release notes (make PR based on 1.2. step)

**Comment by churik** [comment]
@cammellos @jakubgs 
need your review please for phases 1,3,4,5 (all except Testing)
1) should I add/remove something?
2) is it clear?
4) did I mention right people to be pinged?

**Comment by jakubgs** [comment]
### Submission

>8. Prepare PR for F-droid build (instruction TBD)

Instructions are already in place: https://github.com/status-im/status-react/blob/develop/doc/FDROID.md

### Before publishing

> 3. Update Draft with release on https://github.com/status-im/status-react/releases (Jakub)

Not sure what that means. The releases are created automatically by Jenkins. It overwrites them every time the build runs until they stop being drafts. The only thing necessary is to add the link to the post about the release and publish the release.

### At time of publishing

> 1. Add release to DO store and update site (Jakub - requires credentials to upload)

Most of this can be done beforehand, like uploading the builds and pushing the change to https://dev.status.im/ before the final push to prod https://status.im/.

> 4. validate draft release on github release page https://github.com/status-im/status-react/releases (Jakub)

The release can be given the link to the post about it only after marketing pushes it. Other than that everything else should be already in place at the time of publishing.

>5. update https://github.com/status-im/status-react/blob/develop/RELEASES.md with release notes (make PR based on 1.2. step)

I think that's quite backward. With this process if someone goes to the release tag, for example:
https://github.com/status-im/status-react/tree/1.14.0

And they look inside of the `RELEASES.md` file:
https://github.com/status-im/status-react/blob/1.14.0/RELEASES.md

They will find no notes related to `1.14.0`, which is just bad. I think the contents of that file should be actively curated as we make changes to the repo. Or rather, the process should be:

1. Developers make changes in PRs and also always update `CHANGELOG.md` with notable app changes
2. When creating the `release/*` branch the `RELEASES.md` file should be updated based on `CHANGELOG.md`
3. The `develop` branch should also receive changes from `release/*` branch `RELEASES.md` file

What do you think @cammellos ? Would it make sense to maintain a `CHANGELOG.md`?

**Comment by jakubgs** [comment]
This is a nice example of a changelog that is updated with every change:
https://github.com/cortexproject/cortex/blob/master/CHANGELOG.md

**Comment by churik** [comment]
@jakubgs thank you for reviewing!
Updated umbrella issue. 
As for `CHANGELOG.md` - agree it would be nice to use, I used this issue just to describe the process as it is, step by step. And ofc it will be changed with improving release process.


As for place to store it - I suggest to make a template here: https://github.com/status-im/status-react/tree/develop/.github/ISSUE_TEMPLATE and create this issue with every release (will be maintained with person, who will be responsible for current release + me or smbd from test team)

WDYT?
